extra_docs/dataset/README.md

```diff
@@ -0,0 +1,269 @@
+# Datasets Module
+A Dataset is used to manipulate records.
+
+### [Initialize](#init) 
+To use dataset features, just add `dataOperations` module to the client initialization. Notice, that for Node.js
+you have to import `dataOperations` from `jexia-sdk-js/node`, but if you are running your application in a browser, you should import it from `jexia-sdk-js/browser`. That is because file format is different under these two platforms.
+```typescript
+import { jexiaClient, dataOperations } from "jexia-sdk-js/node"; // "jexia-sdk-js/browser" for browser applications
+
+const dataModule = dataOperations();
+
+jexiaClient().init(credentials, dataModule);
+```
+
+### [Inserting records](#inserting-records)
+
+Records can be inserted to a dataset either by sending an array or a single one. The response will always be an array though.
+
+``` Javascript
+[..]
+const posts = dataModule.dataset("posts");
+
+// Multiple records
+const insertQuery = posts.insert([
+  { title: "New Post", content: "content here" },
+  { title: "Another Post", content: "some more content" },
+]);
+
+// Single record
+const insertQuery = posts.insert({
+  title: "New Post",
+  content: "content here",
+});
+
+// Either way, the response will be an array
+insertQuery
+  .execute()
+  .then((records) => {
+    // you will always get an array of created records, including their generated IDs (even when inserting a single record)
+  }).catch((error) => {
+    // you can see the error info here, if something goes wrong
+  });
+[..]
+```
+
+### [Updating records](#updating-records)
+
+The `update` method is used to modify records, it always has to be used with the `.where` method and a filter criteria.
+
+``` Javascript
+[..]
+const filter = field("id").isInArray([1, 2]);
+
+dataModule.dataset("posts")
+  .update({
+    title: "Changing title",
+  })
+  .where(filter)
+  .execute()
+  .then(() => {
+    // the elements that fill the criteria now have been changed
+  }).catch((error) => {
+    // you can see the error info here, if something goes wrong
+  });
+```
+
+### [Deleting records](#deleting-records)
+
+``` Javascript
+[..]
+dataModule.dataset("posts")
+  .delete()
+  .where(field("title").isLike("test"))
```

Comment:
Can we add somewhere that Where statement is mandatory for delete operation, please?